checker/jdk/nullness/src/java/util/concurrent/ConcurrentHashMap.java

```diff
@@ -0,0 +1,42 @@
+package java.util.concurrent;
+import org.checkerframework.dataflow.qual.Pure;
+import org.checkerframework.dataflow.qual.SideEffectFree;
+import java.util.*;
+import org.checkerframework.checker.nullness.qual.KeyFor;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class ConcurrentHashMap<K extends @NonNull Object, V extends @NonNull Object> extends AbstractMap<K, V>

```

Comment:
All the same specifications should also appear in the supertype, `ConcurrentMap`. At the moment there are many more annotations here than in `ConcurrentMap`.